tests/key_field/key_field_sample_generated.h

```diff
@@ -0,0 +1,260 @@
+// automatically generated by the FlatBuffers compiler, do not modify
+
+
+#ifndef FLATBUFFERS_GENERATED_KEYFIELDSAMPLE_KEYFIELD_SAMPLE_H_
+#define FLATBUFFERS_GENERATED_KEYFIELDSAMPLE_KEYFIELD_SAMPLE_H_
+
+#include "flatbuffers/flatbuffers.h"
+
+// Ensure the included flatbuffers.h is the same version as when this file was
+// generated, otherwise it may not be compatible.
+static_assert(FLATBUFFERS_VERSION_MAJOR == 22 &&
+              FLATBUFFERS_VERSION_MINOR == 10 &&
+              FLATBUFFERS_VERSION_REVISION == 26,
+             "Non-compatible flatbuffers version included");
+
+namespace keyfield {
+namespace sample {
+
+struct Baz;
+
+struct Bar;
+
+struct FooTable;
+struct FooTableBuilder;
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(1) Baz FLATBUFFERS_FINAL_CLASS {
+ private:
+  uint8_t a_[4];
+  uint8_t b_;
+
+ public:
+  Baz()
+      : a_(),
+        b_(0) {
+  }
+  Baz(uint8_t _b)
+      : a_(),
+        b_(flatbuffers::EndianScalar(_b)) {
+  }
+  Baz(flatbuffers::span<const uint8_t, 4> _a, uint8_t _b)
+      : b_(flatbuffers::EndianScalar(_b)) {
+    flatbuffers::CastToArray(a_).CopyFromSpan(_a);
+  }
+  const flatbuffers::Array<uint8_t, 4> *a() const {
+    return &flatbuffers::CastToArray(a_);
+  }
+  bool KeyCompareLessThan(const Baz * const o) const {
+    return KeyCompareWithValue(o->a()->data()) < 0;
+  }
+  int KeyCompareWithValue(const uint8_t  _a[4]) const { 
```

Comment:
OK, i double check this, it just resolves to `const uint8_t * _a`, which I think I prefer.

I would make it `const uint8_t * const _a`